improver/nbhood/recursive_filter.py

```diff
@@ -413,48 +375,48 @@ def process(
                 A cubelist containing two cubes of smoothing_coefficient values,
                 one corresponding to smoothing in the x-direction, and the other
                 to smoothing in the y-direction.
-            mask_cube (iris.cube.Cube or None):
-                Cube containing an external mask to apply to the cube before
-                applying the recursive filter.
 
         Returns:
             iris.cube.Cube:
                 Cube containing the smoothed field after the recursive filter
                 method has been applied.
         """
         cube_format = next(cube.slices([cube.coord(axis="y"), cube.coord(axis="x")]))
-        (
-            smoothing_coefficients_x,
-            smoothing_coefficients_y,
-        ) = self._validate_and_pad_coefficients(cube_format, smoothing_coefficients)
+        coeffs_x, coeffs_y = self._validate_coefficients(
+            cube_format, smoothing_coefficients
+        )
+
+        mask_cube = None
+        if np.ma.is_masked(cube_format.data):
+            # assumes mask is the same for each x-y slice
```

Comment:
I think the only time this is unlikely to be true is if we pass in a cube with multiple time points, we aren't planning to do that with IMPROVER, but it is conceivable that other users may want to do this. I suggest adding a trap in this plugin to prevent multiple times, or at least, multiple times with masks.